src/main.cc

```diff
@@ -0,0 +1,84 @@
+
+#include <cstdlib> // For EXIT_SUCCESS, EXIT_FAILURE
+#include <cstring> // For strcmp
+#include <iostream>
+#include <string>
+
+#include "../include/tokenizer.h"
+#include "../include/vvtbi.h"
+
+// VVTBI's version number.
+constexpr const char* VERSION = "2.0";
+
+// The message printed if no file is given.
+const std::string NOARGS = "VERSION: " + std::string(VERSION) +
+                           "\n"
+                           "***************************************\n"
+                           "  Howto: ./vvtbi [-debug] file." +
+                           std::string("vvtb") + "\n";
+
+/**
+ * @brief Check if the filename has the valid extension.
+ *
+ * @param filename File to check against.
+ * @return true if filename ends with the expected extension.
+ */
+bool valid(const std::string& filename) {
+    size_t pos = filename.rfind('.');
+    if (pos == std::string::npos)
+        return false;
+    std::string ext = filename.substr(pos + 1);
+    return ext == "vvtb";
+}
+
+int main(int argc, char** argv) {
+    if (argc < 2) {
+        // No arguments provided; print usage message.
+        std::cout << NOARGS;
+    }
+    // Debug mode: run the tokenizer and print tokens.
+    else if (std::strcmp(argv[1], "-debug") == 0) {
```

Comment:
oh my god i forgot `argv` is type `char **` that is disgusting, uhh ok nvm maybe leave it